src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs

```diff
@@ -64,6 +64,13 @@ private ProcessArgumentBuilder GetArguments(FilePath solution, MSBuildSettings s
                 builder.Append(string.Concat("/p:\"Configuration\"=", configuration.Quote()));
             }
 
+            // Buid for a specific platform?
+            if (settings.PlatformTarget != default(PlatformTarget))
```

Comment:
Ok, I understand what you mean.

The platform target is used to chose what version of MSBuild to be used, and this conflicts.
Since we used `MSIL` as the default platform target, tests are failing.

We could make the PlatformTarget nullable and replace the `GetAlternativeToolPaths` [MSBuildRunner.cs](https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs#L146).

```csharp
protected override IEnumerable<FilePath> GetAlternativeToolPaths(MSBuildSettings settings)
{
    if (settings == null)
    {
        throw new ArgumentNullException("settings");
    }

    var platform = settings.PlatformTarget ?? PlatformTarget.MSIL;
    var path = MSBuildResolver.GetMSBuildPath(_fileSystem, _environment, settings.ToolVersion, platform);
    if (path != null)
    {
        return new[] { path };
    }

    return Enumerable.Empty<FilePath>();
}
```

We would also need to remove the assignment of the default PlatformTarget in the [MSBuildSettings constructor](https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs#L80).

Finally, we should change (and rename) the [test for the default values in MSBuildSettings](https://github.com/cake-build/cake/blob/develop/src/Cake.Common.Tests/Unit/Tools/MSBuild/MSBuildSettingsTests.cs#L22) to reflect these changes:

```csharp
[Fact]
public void Should_Set_Default_Platform_Target_To_Null()
{
    // Given, When
    var settings = new MSBuildSettings();

    // Then
    Assert.Null(settings.PlatformTarget);
}
```

What do you think?